### PR TITLE
Add 'duplicate' to the set theory filters

### PR DESCRIPTION
**Pull request by flatline-studios**
##### SUMMARY

'duplicate' will take a list of values, and return all duplicate values found within it, if any.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

duplicate

##### ADDITIONAL INFORMATION

```paste below
- debug:
    msg: '{{ [1, 1, 1, 2, 3, 3, 4, 5, 6, 7, 8, 8, 8, 8, 8, 8] | duplicate }}'

...

MSG:

[1, 3, 8]
```

### CODE REVIEW

File: lib/ansible/plugins/filter/mathstuff.py [hunk 1]
```diff
@@ -86,6 +86,23 @@ def _do_fail(e):
     return c
 
 
+@environmentfilter
+def duplicate(environment, a):
+    if isinstance(a, Hashable):
```
Comment (sivel) [inline lib/ansible/plugins/filter/mathstuff.py:91]: This `Hashable` check isn't needed.

Additionally, instead of the current code, make use of `collections.Counter`


```
from collections import Counter
@environmentfilter
def duplicate(environment, a):
    return [k for k, v in Counter(a).items() if v > 1]
```

**Review by sivel — CHANGES_REQUESTED** [review]
In addition to the other changes, this needs a changelog fragment (see `changelog/fragments`) and tests (at minimum unit tests, potentially the addition of integration tests too).
